data_creation/data_creation.py

```diff
@@ -0,0 +1,231 @@
+            raise
+
+    def _process_sentence(self, sentence_entry):
+        """
+        Processes a single sentence entry from the DGS Corpus.
+
+        :param sentence_entry: A single sentence entry from the DGS Corpus.
+        :return: Dictionary with sentence ID, updated pose, original pose, and metadata.
+        """
+        sentence = sentence_entry['sentence']
+        glosses = sentence['glosses']['gloss'].numpy()
+        gloss_start_times = sentence['glosses']['start'].numpy()
+        gloss_end_times = sentence['glosses']['end'].numpy()
+
+        pose_datum = sentence_entry['pose']
+        pose_object = self._create_pose_object(pose_datum, "dgs_corpus")
+        pose_data = pose_datum['data'].numpy() # (Frames, People, Points, Dims)
+        fps = pose_datum['fps'].numpy()
+        conf = pose_datum['conf'].numpy()
+        pose_body = NumPyPoseBody(fps, pose_data, conf)
+
+        updated_gloss_poses = []
+        original_gloss_poses = []
+
+
+        # metadata = {
+        #     "sentence_id": sentence['id'].numpy().decode('utf-8'),
+        #     "glosses": [],
+        #     "gloss_start_times": gloss_start_times.tolist(),
+        #     "gloss_end_times": gloss_end_times.tolist()
+        # }
+
+        for gloss, start, end in zip(glosses, gloss_start_times, gloss_end_times):
+            gloss = gloss.decode('utf-8')
+            start_frame = math.floor(start / 1000 * fps)
+            end_frame = math.ceil(end / 1000 * fps)
+            print(f"fps: {fps}")
+            print(f"start_frame: {start_frame}, end_frame: {end_frame}, data_size: {len(pose_body.data)}")
+            ### Error: start_frame: 299, end_frame: 304, data_size: 133
+
+            # Extract original pose for current gloss
+            original_pose_body = pose_body.select_frames(range(start_frame, end_frame))
+            original_pose_header = self._get_pose_header("dgs_corpus")
+            original_pose = Pose(original_pose_header, original_pose_body)
+            original_gloss_poses.append(original_pose)
+
+
+            # Determine if the gloss should be replaced
+            if self._should_replace(gloss):
+                types_gloss_pose_path = self.dictionary[gloss]['views']['pose']
+                with open(types_gloss_pose_path, "rb") as f:
+                    types_gloss_pose = Pose.read(f.read())
+                    gloss_pose = types_gloss_pose
+                self.replaced_signs += 1
+            else:
+                gloss_pose = original_pose
+
+
+            updated_gloss_poses.append(gloss_pose)
+            # metadata["glosses"].append(gloss)
+
+        # Concatenate poses
+        updated_pose = concatenate_poses(updated_gloss_poses)
+        original_pose = concatenate_poses(original_gloss_poses)
+
+        return {
+            "sentence_id": metadata["sentence_id"],
+            "updated_pose": updated_pose,
+            "original_pose": original_pose,
+            "metadata": metadata
+        }
+
```

Comment:
before i review the rest of the code, could you upload an example here?
sentence_id =
updated_poes (save as file, and upload)
original_pose (save as file, and upload)
metadata =

Reply:
Hi Amit, sorry that there is still an issue with the code, as shown below. I'll upload examples after it's fixed. Thank you for your understanding!